reconcile/openshift_resources.py

```diff
@@ -119,107 +175,134 @@ def fetch_provider_resource(path):
     return openshift_resource
 
 
+def fetch_provider_vault_secret(name, path):
+    body = {
+        "apiVersion": "v1",
+        "kind": "Secret",
+        "type": "Opaque",
+        "metadata": {
+            "name": name
+        },
+        "data": {}
+    }
+
+    # get the fields from vault
+    raw_data = vault_client.read_all(path)
+    for k, v in raw_data.items():
+        body['data'][k] = base64.b64encode(v)
+
+    openshift_resource = OR(body)
+
+    try:
+        openshift_resource.verify_valid_k8s_object()
+    except (KeyError, TypeError) as e:
+        k = e.__class__.__name__
+        e_msg = "Invalid data ({}). Skipping resource: {}"
+        raise FetchVaultSecretError(e_msg.format(k, path))
+
+    return openshift_resource
+
+
+def fetch_openshift_resource(resource):
+    provider = resource['provider']
+    path = resource['path']
+
+    if provider == 'resource':
+        try:
```

Comment:
remove the try/except blocks from this function and put them in `fetch_desired_state`

Reply:
fixed